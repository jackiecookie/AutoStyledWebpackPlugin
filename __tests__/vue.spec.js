const webpack = require("webpack")
const path = require("path");
const fs = require("fs");
const rimraf = require("rimraf");
const { GetVueConfig } = require("./config");


const OUTPUT_DIR = "dist";

jest.setTimeout(300000);

describe("test", () => {
  beforeAll(done => {
    rimraf(path.join(__dirname, OUTPUT_DIR), done);
  });

  function testPlugin(name, done, expectModule, injectCssOptions = { style: "style" }) {
    const webpackConfig = GetVueConfig(name, OUTPUT_DIR, injectCssOptions, __dirname)
    let outputFile = `${name}.js`;
    webpack(webpackConfig, (err, state) => {
      if (err) {
        throw err
      }
      console.log('object', state.toString());
      let outputFilePath = path.join(__dirname, OUTPUT_DIR, outputFile);
      const outputFileExists = fs.existsSync(outputFilePath);
      expect(outputFileExists).toBe(true);
      const content = fs.readFileSync(outputFilePath).toString();
      //  expect(content).toMatchSnapshot(name);
      let { exist = [], notExist = [] } = expectModule;
      let modules = state.compilation.modules;
      exist.every(existModule => {
        let index = modules.findIndex(module => {
          return new RegExp(existModule + "$").test(module.request);
        });
        expect(index).toBeGreaterThan(-1);
      });

      notExist.every(notexistModule => {
        let index = modules.findIndex(module => {
          return new RegExp(notexistModule + "$").test(module.request);
        });
        expect(index).toEqual(-1);
      });
      done();
    });
  }


  const library = "./element";
  test("test vue file", done => {
    testPlugin(
      "test",
      done,
      {
        exist: ["button.css"]
      },
      {
        SSR:true,
        library,
        style: function style(rawRequest, name) {
          return `${rawRequest}/style/${name.toLowerCase()}.css`;
        }
      }
    );
  });

});
