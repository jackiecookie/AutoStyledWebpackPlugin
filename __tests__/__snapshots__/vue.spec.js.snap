// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test vue file: test 1`] = `"!function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,\\"default\\",{enumerable:!0,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=1)}([function(e,t){},function(e,t,n){\\"use strict\\";n.r(t);var r=function(){var e=this.$createElement;return(this._self._c||e)(\\"div\\",[this._ssrNode(\\"测试\\")])};r._withStripped=!0;n(0);var o=n(0),i={beforeCreate(){console.log(\\"Button\\")}};i.beforeCreate=[function(e){e=e||this.$vnode&&this.$vnode.ssrContext||this.parent&&this.parent.$vnode&&this.parent.$vnode.ssrContext,o.__inject__&&o.__inject__(e)}];var s=function(e,t,n,r,o,i,s,u){var a,c=\\"function\\"==typeof e?e.options:e;if(t&&(c.render=t,c.staticRenderFns=n,c._compiled=!0),r&&(c.functional=!0),i&&(c._scopeId=\\"data-v-\\"+i),s?(a=function(e){(e=e||this.$vnode&&this.$vnode.ssrContext||this.parent&&this.parent.$vnode&&this.parent.$vnode.ssrContext)||\\"undefined\\"==typeof __VUE_SSR_CONTEXT__||(e=__VUE_SSR_CONTEXT__),o&&o.call(this,e),e&&e._registeredComponents&&e._registeredComponents.add(s)},c._ssrRegister=a):o&&(a=u?function(){o.call(this,this.$root.$options.shadowRoot)}:o),a)if(c.functional){c._injectStyles=a;var f=c.render;c.render=function(e,t){return a.call(t),f(e,t)}}else{var l=c.beforeCreate;c.beforeCreate=l?[].concat(l,a):[a]}return{exports:e,options:c}}(i,r,[],!1,null,null,\\"1c3a13e3\\");s.options.__file=\\"__tests__/support/test.vue\\";t.default=s.exports}]);"`;
